Keen.NET.Test/KeenClientTest.cs

```diff
@@ -46,6 +47,43 @@ public static void ResetEnv()
     }
 
     [TestFixture]
+    public class AccessKeysTest : TestBase
+    {
+        [Test]
+        public void CreateAccessKey_Success()
+        {
+            var settings = new ProjectSettingsProvider(projectId: "X", masterKey: SettingsEnv.MasterKey); // Replace X with respective value
+            var client = new KeenClient(settings);
+
+            if (UseMocks)
+                client.AccessKeys = new AccessKeysMock(settings,
+                    createAccessKey: new Func<AccessKey, IProjectSettings, JObject>((e,p) =>
+                    {
+                        Assert.True(p == settings, "Incorrect Settings");
+                        Assert.NotNull(e.Name, "Expected a name for the newly created Key");
+                        Assert.NotNull(e.Permitted, "Expected a list of high level actions this key can perform");
+                        Assert.NotNull(e.Options, "Expected an object containing more details about the key’s permitted and restricted functionality");
+                        if ((p == settings) && (e.Name == "TestAccessKey") && (e.IsActive) && e.Permitted.First() == "queries" && e.Options.CachedQueries.Allowed.First() == "my_cached_query")
+                            return new JObject(); 
+                        else
+                            throw new Exception("Unexpected value");
+                    }));
+
+            HashSet<string> permissions = new HashSet<string>() { "queries" };
+            List<Core.Query.QueryFilter> qFilters = new List<Core.Query.QueryFilter>() { new Core.Query.QueryFilter("customer.id", Core.Query.QueryFilter.FilterOperator.Equals(), "asdf12345z") };
```

Comment:
If/when you create a new file for Access Key tests, I'd recommend adding `using Keen.Core.Query;` and dispensing with the explicit namespace here.